src/core/common/tlvs.cpp

```diff
@@ -269,16 +269,24 @@ template Error Tlv::AppendUintTlv<uint8_t>(Message &aMessage, uint8_t aType, uin
 template Error Tlv::AppendUintTlv<uint16_t>(Message &aMessage, uint8_t aType, uint16_t aValue);
 template Error Tlv::AppendUintTlv<uint32_t>(Message &aMessage, uint8_t aType, uint32_t aValue);
 
-Error Tlv::AppendTlv(Message &aMessage, uint8_t aType, const void *aValue, uint8_t aLength)
+Error Tlv::AppendTlv(Message &aMessage, uint8_t aType, const void *aValue, uint16_t aLength)
 {
-    Error error = kErrorNone;
-    Tlv   tlv;
-
-    OT_ASSERT(aLength <= Tlv::kBaseTlvMaxLength);
+    Error       error = kErrorNone;
+    ExtendedTlv extTlv;
+    Tlv         tlv;
 
-    tlv.SetType(aType);
-    tlv.SetLength(aLength);
-    SuccessOrExit(error = aMessage.Append(tlv));
+    if (aLength > ot::Tlv::kBaseTlvMaxLength)
```

Comment:
Can remove `ot::Tlv`? (sicne this is a method on the same type).

Reply:
fixed.